mlrun/model_monitoring/application.py

```diff
@@ -284,15 +260,33 @@ def __init__(
         self.output_stream = None
         self.name = name or "PushToMonitoringWriter"
 
-    def do(self, event: list[ModelMonitoringApplicationResult]) -> None:
+    def do(self, event: Tuple[list[ModelMonitoringApplicationResult], dict]) -> None:
         """
         Push application results to the monitoring writer stream.
 
-        :param event: Monitoring result(s) to push.
+        :param event: Monitoring result(s) to push and the original event from the controller.
         """
         self._lazy_init()
-        for result in event:
-            data = result.to_dict()
+        results, origin_event = event
+        metadata = {
+            mm_constant.WriterEvent.APPLICATION_NAME: origin_event[
+                mm_constant.ApplicationEvent.APPLICATION_NAME
+            ],
+            mm_constant.WriterEvent.ENDPOINT_ID: origin_event[
+                mm_constant.ApplicationEvent.ENDPOINT_ID
+            ],
+            mm_constant.WriterEvent.START_INFER_TIME: origin_event[
+                mm_constant.ApplicationEvent.START_INFER_TIME
+            ],
+            mm_constant.WriterEvent.END_INFER_TIME: origin_event[
+                mm_constant.ApplicationEvent.END_INFER_TIME
+            ],
+            mm_constant.WriterEvent.CURRENT_STATS: json.dumps(
+                origin_event[mm_constant.ApplicationEvent.CURRENT_STATS]
+            ),
+        }
+        for result in results:
+            data = result.to_dict().update(metadata)
```

Comment:
This statement sets `data` to `None` - you misuse the `update` method.
https://docs.python.org/3/library/stdtypes.html#dict.update

Reply:
ops thanks:)